src/jinja2/filters.py

```diff
@@ -66,8 +66,7 @@ def make_attrgetter(environment, attribute, postprocess=None, default=None):
     def attrgetter(item):
         for part in attribute:
             item = environment.getitem(item, part)
-
-            if default and isinstance(item, Undefined):
+            if (default or not default) and isinstance(item, Undefined):
```

Comment:
How about `None` as default value here?  it seems `item` will never be Undefined.

it is better to use `default is not None` if None can not use as default value in `map`.

or change the default value to `default=Undefined()` so that None can be use as default value in `map`

Reply:
I have made the change in ```filters.py``` so that it now uses ```default is not None```.